File: 2nd Year/1st Semester/OOP/assignment5/stack.cpp
#include <iostream>
using namespace std;

class Error {
    const char *p;
    public:
        Error(const char *s):p(s) {}
        void disp() {
            cout << p << endl;
        }
};

class Stack {
    int *a, top, size;
    public:
        Stack(int p=0) {
            size=p;
            a = new int[size];
            top=-1;
        }
        void push(int x) {
            if (top==size-1) throw Error("Overflow!");
            top++;
            a[top]=x;
            cout << "Pushed element!\n";
        }
        int pop() {
            if (top==-1) throw Error("Underflow!");
            int tmp = a[top--];
            return tmp;
        }
};

int main() {
    Stack s(3);
    int el[] = {1,3,4, 5, 10};
    try {
        for (int i=0; i<5; i++) {
            s.push(el[i]);
        }
    }
    catch (Error ob) {
        ob.disp();
    }
    try {
        for (int i=0; i<5; i++) {
            cout << s.pop() << endl;
        }
    }
    catch (Error ob) {
        ob.disp();
    }
}

File: 2nd Year/2nd Semester/OOS/assignment_1/q3/stack.java
class Stack {
	/*
 * push, pop, print
 * const to create an array of integers, size supplied by user
 * create stack to hold maximum of 30
 * push 10,20,30,15,9
 * print
 * pop thrice
 * print
 *
 */
	int[] s;
	int n;
	int top=-1;
	Stack(int t) {
		n = t;
		s = new int[t];
	}
	void push(int v) {
		if (top==n-1) {
			System.out.println("Overflow!");
		}
		s[++top] = v;
	}
	int pop() {
		if (top<0) {
			System.out.println("Underflow!");
		}
		top--;
		return s[top];
	}
	void print() {
		System.out.println("The elements are:");
		for (int i = 0; i<=top; i++) {
			System.out.println(s[i]);
		}
	}
}

class Demo {
	public static void main(String args[]) {
		Stack s = new Stack(30);
		s.push(10); s.push(20); s.push(30); s.push(15); s.push(9);
		s.print();
		s.pop();
		s.pop();
		s.pop();
		System.out.println("Popped thrice!");
		s.print();
	}
}

